File: management/controllers/shopinfo.py
#encoding=utf-8
import urllib2
from bs4 import BeautifulSoup
from urlparse import parse_qs
from urlparse import urlparse
import re

def fetch(shopurl):
    html = urllib2.urlopen(shopurl).read()
    soup = BeautifulSoup(html)
    titletag = soup.findChild('p',{'class':'box'}).text
    title = titletag[3:]
    pictag = soup.findChild('td',{'class':'pic'})
    piclink = pictag.img['src']
    nick = title
    if shopurl.find('taobao')>=0:
        ownertag = soup.findChild('td',{"valign":"top"})
        owner = ownertag.next
        nick = owner.strip()[3:]

    wwimg = soup.findChild('img',{'alt':'ww'})
    pgtor = wwimg.parentGenerator()
    sidtag = pgtor.next()['href']
    uparser = urlparse(sidtag)
    shopid = parse_qs(uparser.query)['shopId'][0]
    shopid = int(shopid)
    scoretag = pgtor.next().text
    scorearray = re.findall(ur'\d\.\d分',scoretag)
    item_score = float(scorearray[0][:-1])
    service_score = float(scorearray[1][:-1])
    delivery_score = float(scorearray[2][:-1])

    print title 
    print nick
    print piclink
    print shopid
    print item_score
    print service_score
    print delivery_score


if __name__=="__main__":
    fetch("http://shop71839143.m.taobao.com")

File: management/controllers/shopinfo.go
package controllers

import (
	"github.com/PuerkitoBio/goquery"
	"github.com/jason-zou/taobaosdk/rest"
	"log"
	"net/url"
	"regexp"
	"strconv"
	"strings"
)

func fetch(link string) (*rest.Shop, error) {
	doc, e := goquery.NewDocument(link)
	if e != nil {
		log.Println(e.Error())
		return nil, e
	}

	titletag := doc.Find("p.box").Text()
	title := titletag[7:]
	title = strings.TrimSpace(title)
	log.Println(title)
	pictag := doc.Find("td.pic img")
	piclink, _ := pictag.Attr("src")
	wwimg := doc.Find("img[alt=ww]")
	src, _ := wwimg.Attr("src")
	srcparse, _ := url.Parse(src)
	srcquery := srcparse.Query()
	nick := srcquery.Get("nick")
	log.Println(nick)
	sidtag, exists := wwimg.Parent().Attr("href")
	var sid int
	if exists {
		URL, _ := url.Parse(sidtag)
		query := URL.Query()
		sid, _ = strconv.Atoi(query.Get("shopId"))
	}
	detail := wwimg.Parent().Parent().Text()
	re := regexp.MustCompile("\\d\\.\\d分")
	scores := re.FindAllString(detail, -1)
	var item_score, service_score, delivery_score float32
	item_score2, _ := strconv.ParseFloat(scores[0][0:len(scores[0])-3], 32)
	service_score2, _ := strconv.ParseFloat(scores[1][0:len(scores[0])-3], 32)
	delivery_score2, _ := strconv.ParseFloat(scores[2][0:len(scores[0])-3], 32)
	service_score = float32(service_score2)
	item_score = float32(item_score2)
	delivery_score = float32(delivery_score2)
	var shopscore *rest.ShopScore
	deliver := &rest.RateScore{Score: delivery_score}
	desc := &rest.RateScore{Score: item_score}
	service := &rest.RateScore{Score: service_score}
	kps := &rest.ShopKPS{DeliveryScore: deliver, DescScore: desc, ServiceScore: service}
	shopscore.SemiScore = kps
	shopinfo := &rest.Shop{Nick: nick, Title: title, PicPath: piclink, ShopScore: shopscore, Sid: sid}
	return shopinfo, nil
}
